modules/light-clients/08-wasm/types/migrate_contract_test.go

```diff
@@ -0,0 +1,140 @@
+package types_test
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
+	wasmtesting "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (suite *TypesTestSuite) TestMigrateContract() {
+	var (
+		oldHash        [32]byte
+		newHash        [32]byte
+		payload        []byte
+		expClientState *types.ClientState
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success: new and old code hash are different",
+			func() {
+				newHash = sha256.Sum256([]byte{1, 2, 3})
+				err := ibcwasm.CodeHashes.Set(suite.chainA.GetContext(), newHash[:])
+				suite.Require().NoError(err)
+
+				payload = []byte{1}
+				expChecksum := wasmvmtypes.ForceNewChecksum(hex.EncodeToString(newHash[:]))
```

Comment:
Why do we need this? Cannot we use `newHash` directly in line 42?

Reply:
Because wasmvm wraps it in this way